test/functional/test-intersection-observer.js

```diff
@@ -25,31 +25,45 @@ import * as sinon from 'sinon';
 
 
 describe('getIntersectionChangeEntry', () => {
+  let sandbox;
+  beforeEach(() => {
+    sandbox = sinon.sandbox.create();
+    sandbox.useFakeTimers();
+  });
+
+  afterEach(() => {
+    sandbox.restore();
+  });
 
   it('intersect correctly base', () => {
-    const time = 123;
     const rootBounds = layoutRectLtwh(0, 100, 100, 100);
     const layoutBox = layoutRectLtwh(50, 50, 150, 200);
-    const change = getIntersectionChangeEntry(time, rootBounds, layoutBox);
+    const change = getIntersectionChangeEntry(layoutBox, null, rootBounds);
 
     expect(change).to.be.object;
-    expect(change.time).to.equal(123);
+    expect(change.time).to.equal(Date.now());
 
-    expect(change.rootBounds).to.equal(rootBounds);
-    expect(change.rootBounds.x).to.equal(0);
-    expect(change.rootBounds.y).to.equal(100);
-    expect(change.boundingClientRect).to.jsonEqual({
+    expect(change.rootBounds).to.deep.equal({
+      'left': 0,
+      'top': 100,
+      'width': 100,
+      'height': 100,
+      'bottom': 200,
+      'right': 100,
+      'x': 0,
+      'y': 100,
+    });
+    expect(change.boundingClientRect).to.deep.equal({
       'left': 50,
-      'top': -50,
```

Comment:
Well, besides it not accounting for the owner element. But no, `intersectionRect` had the correct implementation.